src/main/java/seedu/ddd/model/Model.java

```diff
@@ -85,7 +86,7 @@ public interface Model {
      * Deletes the given person.
      * The person must exist in the address book.
      */
-    void deleteContact(Contact target);
+    void deleteContact(Contact target) throws CommandException;
```

Comment:
I don't we should throw a `CommandException` from `Model` because it breaks abstraction. I think we can do this check in `DeleteContact` instead.